File: python/字符串/387. 字符串中的第一个唯一字符.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        from collections import defaultdict
        d = defaultdict(int)
        for i in s:
            d[i]+=1
        for i,v in enumerate(s):
            if d[v] == 1:
                return i
        return -1

File: go/string/387. 字符串中的第一个唯一字符.go
package string

func firstUniqChar(s string) int {
	d := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		d[s[i]]++
	}
	for i := 0; i < len(s); i++ {
		if d[s[i]] == 1 {
			return i
		}
	}
	return -1
}
